pkg/cloud/devicemanager/manager.go

```diff
@@ -119,7 +119,7 @@ func (d *deviceManager) NewDevice(instance *ec2.Instance, volumeID string) (*Dev
 
 	// Check if this volume is already assigned a device on this machine
 	if path := d.getPath(inUse, volumeID); path != "" {
-		return d.newBlockDevice(instance, volumeID, path, true), nil
+		return nil, fmt.Errorf("volume already assigned")
```

Comment:
Can we add a warning here?
Something like klog.Warningf("Got assignment call for already-assigned volume: %s@%s, volumeID, instanceID)
And if we can get the volume status for free, it would be better to log the volume status as well (just like in-tree)

Reply:
One thing related to ControllerPublishVolume idempotent, I think we'd better keep tracking the volume status, per discussion [here](https://github.com/kubernetes/kubernetes/issues/77389), we can only return this error when volume is in detaching status, and if volume is in attached status, we should return the old statement to keep as much idempotent as we can.. LMK you opinion.